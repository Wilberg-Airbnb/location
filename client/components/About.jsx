import React from "react";
import ReactDOM from "react-dom";
import Map from './Map.jsx';
const About = ({children, onClose, open}) => 
  
  open ?
  ReactDOM.createPortal(
    <div className='modal'>
      <div className='overlay' style={{    
      position: 'fixed',
      top: '0',
      background: 'rgba(0,0,0, 0.4)',
      width: '100%',
      height: '100%',
      left: '0',
    }}>
        <div className='content' style={
          { width: '90%',
            background: 'white',
            padding: '20px',
            position: 'absolute',
            top: '50%',
            left: '50%',
            transform: 'translate(-50%, -50%)'
        }}>
          <div className='modal_close' onClick={onClose}>&times;</div>
          <div>Map should be here</div>
        </div>
     </div>
  </div>,
  document.getElementById('modal'))
  : null


export default About;