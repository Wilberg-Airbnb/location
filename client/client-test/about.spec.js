import React from 'react';
import renderer from 'react-test-renderer';
import About from '../components/About.jsx';


import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
configure({ adapter: new Adapter() });
 
describe('Renders Components', () => {
  let component;
  let closeButton;
  beforeEach( () => {
    component = shallow(<About />)
   
  });

  test("should render initial layout", () => {
    const component = shallow(<About />);
    expect(component.getElements()).toMatchSnapshot();
  });

  test("should create a About modal in component state", ()=> {
    expect(component.contains(<About />)).toBe(false)
  });

  test("should close modal when clicked", ()=> {
    let overlay = component.find('#modal').find('.overlay');
    console.log(component);
    // expect(overlay.length).to.equal(1)

  });

  xtest("should close modal when clicked", ()=> {
    closeButton = component
      .find('.overlay')
      .find('.content')
      .find('#modal_close');
 
    // expect(closeButton.length).to.equal(1)
    closeButton.simulate('click');
    expect(component.state.showModal).toBe(false);
  });

 
});


