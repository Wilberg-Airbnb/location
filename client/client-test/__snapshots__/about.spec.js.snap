// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Components should render initial layout 1`] = `
Array [
  null,
]
`;
